src/main/java/net/sf/mzmine/parameters/parametertypes/filenames/FileNameParameter.java

```diff
@@ -99,7 +99,9 @@ public FileNameParameter cloneParameter() {
     public void setValueFromComponent(FileNameComponent component) {
         File compValue = component.getValue();
         if (extension != null) {
-            if (!compValue.getName().endsWith(extension))
+            if (!(compValue.getName().toUpperCase()
+                    .endsWith(extension.toUpperCase()) || compValue.getName()
+                    .toLowerCase().endsWith(extension.toLowerCase())))
```

Comment:
Well, no. I did not. Here is what I want:

* "file.EXT" remains "file.EXT"
* "file.ext" remains "file.ext"
* "file.ExT" becomes "file.ExT.ext"

Because extension case shouldn't be mixed up, the latter requires the extension be: `not("EXT" or "ext")` to be appended. I feel like I never wrote a line of code if the second proposition is not correct ;) ...

But maybe you find it useless to cover the third case, which is an argument I can eventually share (in which case your line is correct).